lib/tokenize.cpp

```diff
@@ -6225,6 +6225,29 @@ void Tokenizer::simplifyHeadersAndUnusedTemplates()
     for (Token *tok = list.front(); tok; tok = tok->next()) {
         const bool isIncluded = (tok->fileIndex() != 0);
 
+        // Remove constructor initializer
+        if (isIncluded && !mSettings.checkHeaders && Token::Match(tok, "%name% (")) {
+            Token *start = tok->next()->link();
+            if (start) {
+                start = start->next();
+            }
+            if (start && start->str() == ":") {
+                const Token *next = start;
+                while (Token::Match(next, "[,:]")) {
+                    // Skip to ( or {
+                    while (next && !next->link())
```

Comment:
I don't want to skip any random tokens. something more like:

    next = next->next();
    while (Token::Match(next, "%name%|::|<")) {
        if (next->str() == "<")
             // find end token.. I believe there is a utility function in template simplifier
        next = next->next();
    }
    if (!Token::Match(next, "(|{"))
        // not initializer list..
    next = next->link()->next(); // <- goto "," or "{"

Reply:
Looking better at it, If after the previous change '_Token::Match(next, "[,:] %name%")_' The symbol after %name% should always be a link in a constructor initializer list. If not is is something else. %name% should always be a member of delegate constructor.

Only thing now failing is a parameter pack expansion in an initializer list

template<class... Mixins>
class X : public Mixins...
{
public:
    X(const Mixins&... mixins) : Mixins(mixins)... {}
};